manifests/file.pp

```diff
@@ -1,85 +1,53 @@
-# @example
-#   if $::memorysize_mb < 24576.0 {
-#     $max_heap_size_in_mb = floor($::memorysize_mb / 2)
-#   } elsif $::memorysize_mb < 8192.0 {
-#     $max_heap_size_in_mb = floor($::memorysize_mb / 4)
-#   } else {
-#     $max_heap_size_in_mb = 8192
-#   }
-#
-#   $heap_new_size = $::processorcount * 100
+# @summary A defined type for altering files relative to the configuration directory.
 #
-#   cassandra::file { "Set Java/Cassandra max heap size to ${max_heap_size_in_mb}.":
+# @example Set max_heap_size in cassandra-env.sh
+#   cassandra::file { "Set Java/Cassandra max heap size to 500.":
 #     file       => 'cassandra-env.sh',
 #     file_lines => {
 #       'MAX_HEAP_SIZE' => {
-#         line  => "MAX_HEAP_SIZE='${max_heap_size_in_mb}M'",
+#         line  => 'MAX_HEAP_SIZE=500M',
 #         match => '^#?MAX_HEAP_SIZE=.*',
 #       },
 #     }
 #   }
 #
-#   cassandra::file { "Set Java/Cassandra heap new size to ${heap_new_size}.":
+# @example Set heap_newsize in cassandra-env.sh
+#   cassandra::file { "Set Java/Cassandra heap new size to 300.":
 #     file       => 'cassandra-env.sh',
 #     file_lines => {
 #       'HEAP_NEWSIZE'  => {
-#         line  => "HEAP_NEWSIZE='${heap_new_size}M'",
+#         line  => 'HEAP_NEWSIZE=300M',
 #         match => '^#?HEAP_NEWSIZE=.*',
 #       }
 #     }
 #   }
-#   $tmpdir = '/var/lib/cassandra/tmp'
-#
-#   file { $tmpdir:
-#     ensure => directory,
-#     owner  => 'cassandra',
-#     group  => 'cassandra',
-#   }
 #
+# @example Set java.io.tmpdir in jvm.options
 #   cassandra::file { 'Set java.io.tmpdir':
 #     file       => 'jvm.options',
 #     file_lines => {
 #       'java.io.tmpdir' => {
-#         line => "-Djava.io.tmpdir=${tmpdir}",
+#         line => '-Djava.io.tmpdir=/var/lib/cassandra/tmp',
 #       },
 #     },
-#     require    => File[$tmpdir],
 #   }
+#
+# @param file_lines
+#   will be used to create an array of [file_line]
+# @param file
+#   Name of the file relative to `cassandra::config_path`.
+#
 define cassandra::file (
-  $file             = $title,
-  $config_path      = $cassandra::config_path,
-  $file_lines       = undef,
-  $service_refresh  = true,
+  Hash $file_lines,
+  String[1] $file = $title,
 ) {
   include cassandra
-  include cassandra::params
-  include stdlib
-
-  $path = "${config_path}/${file}"
 
-  if $file_lines != undef {
-    if $service_refresh {
-      $default_file_line = {
-        path    => $path,
-        require => Package['cassandra'],
-        notify  => Service['cassandra'],
-      }
-    } else {
-      $default_file_line = {
-        path    => $path,
-        require => Package['cassandra'],
-      }
-    }
-
-    create_resources(file_line, $file_lines, $default_file_line)
+  if $cassandra::service_refresh {
+    $_notify = Service[$cassandra::service_name]
+  } else {
+    $_notify = undef
   }
+
+  create_resources(file_line, $file_lines, { 'path' => "${$cassandra::config_path}/${file}", 'notify' => $_notify })
```

Comment:
Please don't use the function. It messes with the resource ordering, this pattern is deprecated. Instead, please loop on file_lines:
```puppet
$file_lines.each |$line| {
  file_line { $line:
    * => { 'path' => "${$cassandra::config_path}/${file}", 'notify' => $_notify },
  }
}
```

(untested, but something like that)